crates/services/compression/src/service.rs

```diff
@@ -213,28 +327,41 @@ where
         }
     }
 
-    async fn shutdown(self) -> anyhow::Result<()> {
+    async fn shutdown(mut self) -> anyhow::Result<()> {
+        // gracefully handle all the remaining blocks in the stream and then stop
+        while let Some(block_with_metadata) = self.block_stream.next().await {
```

Comment:
I wonder if it would be nice to have a synchronous `.flush_remaining()` method on block_stream or something similar here. From this code, it's not apparent that this shutdown will ever terminate, if `block_stream` just continues to produce blocks.

Reply:
that is an invariant we need to think of - if the `block_stream` continues to produce blocks, is it not the duty of the compression service to make sure they are compressed? as longs as blocks live in the stream, there should be compression performed on them afaiu